applications/openshift/accounts/accounts_no_clusterrolebindings_default_service_account/rule.yml

```diff
@@ -0,0 +1,54 @@
+documentation_complete: true
+
+prodtype: ocp4
+
+title: 'Ensure no ClusterRoleBindings set for default Service Account'
+
+description: |-
+    Using the <tt>default</tt> service account prevents accurate application
+    rights review and audit tracing. Instead of <tt>default</tt>, create
+    a new and unique service account and associate the required ClusterRoleBindings. 
+
+rationale: |-
+    Kubernetes provides a default service account which is used by
+    cluster workloads where no specific service account is assigned to the pod.
+    Where access to the Kubernetes API from a pod is required, a specific service account
+    should be created for that pod, and rights granted to that service account.
+    This increases auditability of service account rights and access making it
+    easier and more accurate to trace potential malicious behaviors to a specific
+    service account and project.
+
+severity: medium
+
+identifiers: {}
+
+references:
+    bsi: APP.4.4.A9
+
+{{% set jqfilter = '[.items[] | select ( .subjects[]?.name == "default" ) | select(.subjects[].namespace | startswith("kube-") or startswith("openshift-") | not) | .metadata.name ] | unique' %}}
+
+ocil_clause: 'default service account is given permissions using ClusterRoleBindings'
+
+ocil: |-
+    Run the following command to retrieve a list of ClusterRoleBindings that are
+    associated to the default service account:
+    <pre>$ oc get clusterrolebindings -o json | jq '{{{ jqfilter }}}'</pre>
+    There should be no ClusterRoleBindings associated with the the default service account 
+    in any namespace.
+
+warnings:
+- general: |-
+    {{{ openshift_filtered_cluster_setting({'/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?limit=10000': jqfilter}) | indent(4) }}}
```

Comment:
On other rules there is a default limit of 500 objects. Such limit feels too low for ClusterRoleBindings and RoleBindings, that can easily exceed 500 even in small clusters. I chose 10000 as a limit here, that should be more suitable for most environments while still keeping the data volume reasonable, because ClusterRoleBindings and RoleBindings are fairly small objects.

Reply:
Perhaps we can explore some ways to make this configurable so that we can reuse the original rule.

Reply:
Oh - actually, disregard my comment. I was under the impression this rule existed somewhere else but I was mistaken.

Reply:
I guess the initial idea of a limit on the API was to reduce api impact from compliance operator runs. I do like to have this configurable, to give users a choice in their decision "impact vs. precision". This also allows a customer having an impact from the rule to be able to act (instead of just disabling the rule, be able to reduce the impact, while still detecting "some" cases).

On the other hand, the limits could create "flashy" results on clusters which might also be adjusted using the limit variable.